build-logic/documentation/src/main/groovy/gradlebuild/docs/ReleaseNotesTransformer.java

```diff
@@ -83,57 +85,73 @@ private Reader transform(Reader in) throws IOException {
                 append("<link rel='stylesheet' type='text/css' href='https://assets.gradle.com/lato/css/lato-font.css'/>");
         addCssToHead(document);
         addJavascriptToHead(document);
-
+        addHighlightJsToHead(document);
         wrapH2InSectionTopic(document);
-        addAnchorsForHeadings(document);
         document.body().prepend("<h1>Gradle Release Notes</h1>");
         addTOC(document);
         wrapContentInContainer(document);
+        removeLeftoverComments(document);
+        dealWithVideos(document);
 
+        return new StringReader(document.toString());
+    }
+
+    private void dealWithVideos(Document document) {
```

Comment:
🤔  Not everything in this method has to do with videos.

I suggest we refactor this a bit. Split this method into separate functions of the form `String doStep(String html)` with appropriate names. (If we ever decide to, it's also easier to test). Then have the "main" method that joins it all together:
```
String rewritten = document.body().html();
rewritten = step1(rewritten);
// ...
document.body().html(rewritten);
```

Reply:
Fixed but I just split up into two different, entirely separate methods.